File: 557-reverse-words-in-a-string-iii/reverse-words-in-a-string-iii.c
void reverse(char *s,int left,int right)
{
    int i,j;
    char temp;
    for(i=left,j=right;i<j;i++,j--)
    {
        temp=s[i];
        s[i]=s[j];
        s[j]=temp;
    }
}

char* reverseWords(char* s)
{
    int left,right;
    int l=0;
    while(s[l]!= '\0')
        l++;
    left=0;
    while(left<l)
    {
        right=left+1;
        while(s[right] != ' ' && s[right] != '\0')
        {
            right++;
        }
        reverse(s,left,right-1);
        left=right+1;
    }
    return s;
}

File: 557-reverse-words-in-a-string-iii/reverse-words-in-a-string-iii.java
class Solution {
    public String reverseWords(String s) {
        char[] str = s.toCharArray();
        int i=0, j=0;
        int n = s.length();
        while(i < n) {
            while(j < n && str[j] != ' ') {
                j++;
            }
            reverse(str, i, j - 1);
            j++;
            i=j;
        }
        return String.valueOf(str);
    }
    private void reverse(char[] str, int left, int right) {
        while (left < right) {
            char temp = str[left];
            str[left++] = str[right];
            str[right--] = temp;
        }
    }
}